crates/router/src/connector/iatapay.rs

```diff
@@ -116,22 +116,33 @@ impl ConnectorCommon for Iatapay {
         res: Response,
         event_builder: Option<&mut ConnectorEvent>,
     ) -> CustomResult<ErrorResponse, errors::ConnectorError> {
-        let response: iatapay::IatapayErrorResponse = res
-            .response
-            .parse_struct("IatapayErrorResponse")
-            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
-
-        event_builder.map(|i| i.set_error_response_body(&response));
-        router_env::logger::info!(connector_response=?response);
-
-        Ok(ErrorResponse {
-            status_code: res.status_code,
-            code: response.error,
-            message: response.message,
-            reason: response.reason,
-            attempt_status: None,
-            connector_transaction_id: None,
-        })
+        let response_error_message = if res.response.is_empty() && res.status_code == 401 {
```

Comment:
In our case we are getting empty string as response body. So this won't be empty. And also could you confirm whether connector is responding with empty body in case of 401 in sbx?

Reply:
No in case of 401 iatapay send empty response body. Tested screenshot has been updated in PR description.